src/ruby/yast/rspec/scr.rb

```diff
@@ -0,0 +1,84 @@
+require "rspec"
+require "yast"
+
+# RSpec extension to handle several agent operations.
+#
+# @example usage
+#
+# require 'yast/rspec/scr'
+#
+# RSpec.configure do |c|
+#   c.include Yast::RSpec::SCR
+# end
+#
+# describe YaST::SCR do
+#   before do
+#     chroot = File.join(File.dirname(__FILE__), "test_chroot")
+#     change_scr_root(chroot)
+#   end
+#
+#   after do
+#     reset_scr_root
+#   end
+#
+#   describe "#Read" do
+#     it "works with the .proc.meminfo path"
+#       # This uses the #path helper from SCRStub and
+#       # reads from ./test_chroot/proc/meminfo
+#       values = Yast::SCR.Read(path(".proc.meminfo"))
+#       expect(values).to include("key" => "value")
+#     end
+#   end
+# end
+module Yast
+  module RSpec
+    module SCR
+      # Shortcut for generating Yast::Path objects
+      #
+      # @param route [String] textual representation of the path
+      # @return [Yast::Path] the corresponding Path object
+      def path(route)
+        Yast::Path.new(route)
+      end
+
+      # Encapsulates subsequent SCR calls into a chroot.
+      #
+      # Raises an exception if something goes wrong.
+      #
+      # @param [#to_s] directory to use as '/' for SCR calls
+      def change_scr_root(directory)
+        # On first call, store the default handler in the stack
+        @scr_handles ||= [Yast::WFM.SCRGetDefault]
```

Comment:
The current implementation is actually a stack, so it would support any level of nested chrooting. We can asume that we never want such thing and drop the ```@scr_handles``` array in favor of two scalars ```@scr_handle``` and ```@previous_scr_handle``` so we raise an exception if ```change_scr_root``` is called twice without calling ```reset_scr_root``` in the meantime.

I don't get the hook part of your comment, can you elaborate?